pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_inverted_index failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_inverted_index [failed](https://mesolite.cluster.engflow.com/invocations/default/94c99940-e492-4c38-ba93-b8437e35228c?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on master @ [67f33c276e3971c77d89f3e1b8c7e803ca61cafd](https://github.com/cockroachdb/cockroach/commits/67f33c276e3971c77d89f3e1b8c7e803ca61cafd):


```
=== RUN   TestLogic_inverted_index
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestLogic_inverted_index1677943844
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    logictestbase.go:577: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports schema-locked-disabled (no issue given)
[22:19:00] --- progress: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index: 89 statements
[22:19:04] --- done: /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index with config fakedist-disk: 369 tests, 1 failures
[22:19:05] --- total progress: 369 statements
--- total: 369 tests, 1 failures
    logic.go:4679: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_inverted_index1677943844
--- FAIL: TestLogic_inverted_index (7.81s)
=== RUN   TestLogic_inverted_index/overlaps_with_array
    logic.go:2339: 
         
        /mnt/engflow/worker/work/1/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index:2141: 
        expected "geos error: IllegalArgumentException: BufferOp::getResultGeometry distance must be a finite value", but no error occurred
--- FAIL: TestLogic_inverted_index/overlaps_with_array (0.69s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_inverted_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-55237

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_inverted_index [failed](https://mesolite.cluster.engflow.com/invocations/default/cd5e41f7-257b-413c-9b30-dbfd4617a94d?testReportRun=1&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc3FsL2xvZ2ljdGVzdC90ZXN0cy9mYWtlZGlzdC1kaXNrOmZha2VkaXN0LWRpc2tfdGVzdA==) on master @ [5cda21f8745d4584fa16c82b4415cdb4c82cc67f](https://github.com/cockroachdb/cockroach/commits/5cda21f8745d4584fa16c82b4415cdb4c82cc67f):


```
=== RUN   TestLogic_inverted_index
    test_log_scope.go:171: test logs captured to: outputs.zip/logTestLogic_inverted_index2712843053
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:141: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
[04:16:14] setting distsql_workmem='86245B';
    logictestbase.go:577: statement/query skipped with reason: unsupported configuration fakedist-disk, statement/query only supports schema-locked-disabled (no issue given)
[04:16:16] --- progress: /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index: 27 statements
[04:16:20] --- done: /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index with config fakedist-disk: 369 tests, 1 failures
[04:16:21] --- total progress: 369 statements
--- total: 369 tests, 1 failures
    logic.go:4679: -- test log scope end --
test logs left over in: outputs.zip/logTestLogic_inverted_index2712843053
--- FAIL: TestLogic_inverted_index (8.03s)
=== RUN   TestLogic_inverted_index/overlaps_with_array
    logic.go:2339: 
         
        /mnt/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/inverted_index:2141: 
        expected "geos error: IllegalArgumentException: BufferOp::getResultGeometry distance must be a finite value", but no error occurred
--- FAIL: TestLogic_inverted_index/overlaps_with_array (0.70s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_inverted_index.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/sql/logictest/testdata/logic_test/inverted_index
+++ b/pkg/sql/logictest/testdata/logic_test/inverted_index
@@ -2137,6 +2137,11 @@ INSERT INTO t151995 (c0, c1) VALUES (
 statement ok
 ANALYZE t151995;
 
+# Clear the stat cache so that the new statistic is guaranteed to be used (which
+# are needed for the inverted join to be picked).
+statement ok
+SELECT crdb_internal.clear_table_stats_cache();
+
 statement error geos error: IllegalArgumentException: BufferOp::getResultGeometry distance must be a finite value
 SELECT *
   FROM t151995 AS t1

